### PR TITLE
[ConstraintElim] Add facts for llvm.abs >= 0

### PR DESCRIPTION
**Pull request by alexander-shaposhnikov**
Add facts for llvm.abs >= 0.

https://alive2.llvm.org/ce/z/GXnMHu

### CODE REVIEW

**Comment by alexander-shaposhnikov** [comment]
@fhahn  - I've just rebased the patch

**Comment by alexander-shaposhnikov** [comment]
gentle ping @fhahn

**Comment by alexander-shaposhnikov** [comment]
@fhahn - what would you say to submitting this ?

File: llvm/lib/Transforms/Scalar/ConstraintElimination.cpp [hunk 1]
```diff
@@ -1730,7 +1730,10 @@ static bool eliminateConstraints(Function &F, DominatorTree &DT, LoopInfo &LI,
     if (!CB.isConditionFact()) {
       Value *X;
       if (match(CB.Inst, m_Intrinsic<Intrinsic::abs>(m_Value(X)))) {
-        // TODO: Add CB.Inst >= 0 fact.
+        // If is_int_min_poison is true then we may assume llvm.abs >= 0.
+        if (!cast<ConstantInt>(CB.Inst->getOperand(1))->isZero())
```
Comment (nikic) [inline llvm/lib/Transforms/Scalar/ConstraintElimination.cpp:1734]:
```suggestion
        if (cast<ConstantInt>(CB.Inst->getOperand(1))->isOne())
```

**Review by nikic — APPROVED** [review]
LGTM

File: llvm/lib/Transforms/Scalar/ConstraintElimination.cpp (hunk 1 above)
Comment (alexander-shaposhnikov) [reply llvm/lib/Transforms/Scalar/ConstraintElimination.cpp]: Done

**Pull request merged**